src/connectors/query-rules/connectQueryRules.ts

```diff
@@ -18,10 +18,12 @@ import {
   NumericRefinement as InternalNumericRefinement,
 } from '../../lib/utils/getRefinements';
 
+type TrackedFilterRefinement = string | number | boolean;
+
 export type ParamTrackedFilters = {
   [facetName: string]: (
-    facetValues: Array<string | number>
-  ) => Array<string | number>;
+    facetValues: TrackedFilterRefinement[]
```

Comment:
I think maybe this parameter should be optional. We want to consider this valid:
```ts
{ genre: () => ['Comedy'] }
```

Reply:
That's a good point!

Reply:
Actually that's wrong. This function gets called with the computed tracked filters (it's never `undefined`).